### PR TITLE
Switch from Ubuntu 16.04 to 18.04. Add forticlient deb file directly …

### PR DESCRIPTION
**Pull request by Enoxime**
…into dockerfile. Add docker-compose file to automate the setup (only need to add your credentials and do a 'docker-compose run'.

It's not tested in an Apple environment.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,14 +1,26 @@
...
+  iptables \
+  && apt-get clean -q && apt-get autoremove --purge \
+  && rm -rf /var/lib/apt/lists/*
 
 WORKDIR /root
 
 # Install fortivpn client unofficial .deb
-RUN wget 'https://hadler.me/files/forticlient-sslvpn_4.4.2329-1_amd64.deb' -O forticlient-sslvpn_amd64.deb
-RUN dpkg -x forticlient-sslvpn_amd64.deb /usr/share/forticlient
+COPY forticlient-sslvpn_amd64.deb .
```
Comment (godric) [inline Dockerfile]: I actually think of it as a feature that the Dockerfile does not depend on a local `.deb`.
If this is an attempt at removing the version number from the Dockerfile, maybe could use an ENV var for the version/url (with a default value).

File: Dockerfile [hunk 2]
```diff
@@ -1,14 +1,26 @@
...
+RUN dpkg -x forticlient-sslvpn_amd64.deb /usr/share/forticlient && rm forticlient-sslvpn_amd64.deb
```
Comment (godric) [inline Dockerfile:25]: I don't see the point to this as the file will still be present in the previous layer, which will count toward the image size.

If you're looking to save some storage space, AFIK the only options are:
 - Downloading the file and deleting it on the same `RUN` command (see previous comment)
 - [Multi-stage builds](https://docs.docker.com/develop/develop-images/multistage-build/) (add a lot of complexity)
 - Buildkit's [new features](https://docs.docker.com/develop/develop-images/build_enhancements/) (still experimental)

Then again the ~5Mb of the package might not event be worth the effort.

File: Dockerfile [hunk 3]
```diff
@@ -17,4 +29,4 @@ RUN /usr/share/forticlient/opt/forticlient-sslvpn/64bit/helper/setup.linux.sh 2
 COPY forticlient /usr/bin/forticlient
 COPY start.sh /start.sh
 
-CMD [ "/start.sh" ]
+CMD [ "/start.sh" ]
```
Comment (godric) [inline Dockerfile]: We want the newlines at the end of files.

File: forticlient [hunk 4]
```diff
@@ -33,4 +33,4 @@ expect {
   eof {
     exit
   }
-}
+}
```
Comment (godric) [inline forticlient]: We want the newlines at the end of files.

File: start.sh [hunk 5]
```diff
@@ -16,4 +16,4 @@ while [ true ]; do
   /usr/bin/forticlient
   echo "------------ VPN exited ------------"
   sleep 10
-done
+done
```
Comment (godric) [inline start.sh]: We want the newlines at the end of files.

**Review by godric — CHANGES_REQUESTED** [review]
First of all, thank you for the PR, it is really appreciated. Especially the Dockerfile cleanup and the addition of the docker-compose file.

See my comments for a few change requests,
One more thing that I couldn't comment on the .deb, but adding a binary file to the repository is not an option, if you really want to copy the file and not have the build download it, please provide a shell script or something to download it from the build host.

File: Dockerfile (hunk 2 above)
Comment (Enoxime) [reply Dockerfile:25]: You are right! I didn't know about the build new features you mentioned. Pretty interresting thought.
I would suggest to use the multi-stage builds but as you mentioned earlier, It doesn't worth the effort.

**Comment by Enoxime** [comment]
Rollback for the newline ending. And all other changes required are done.

**Review by godric — CHANGES_REQUESTED** [review]
Excellent, I think we're almost there,
Unfortunately It won't build now because wget can't validate the root certificate for https://hadler.me:
```
Step 6/11 : RUN wget 'https://hadler.me/files/forticlient-sslvpn_4.4.2329-1_amd64.deb' -O forticlient-sslvpn_amd64.deb
 ---> Running in f2c880e964e1
--2019-08-30 12:03:41--  https://hadler.me/files/forticlient-sslvpn_4.4.2329-1_amd64.deb
Resolving hadler.me (hadler.me)... 77.235.64.70
Connecting to hadler.me (hadler.me)|77.235.64.70|:443... connected.
ERROR: cannot verify hadler.me's certificate, issued by 'CN=Let\'s Encrypt Authority X3,O=Let\'s Encrypt,C=US':
  Unable to locally verify the issuer's authority.
To connect to hadler.me insecurely, use `--no-check-certificate'.
```

This is easily fixed by adding the `ca-certificates` ubuntu package, that must have been included before as a dependency of `curl`

On the side I'll have to see if we can setup some Continuous Integration to at least check that the image build and starts OK, cc @mrjmad

**Comment by Enoxime** [comment]
My bad. I forget to test it.
I added the ca-certificates and test it with the docker-compose file.

**Comment by godric** [comment]
Okay Let's go!
Thanks you again for your contribution!

**Review by godric — APPROVED** [review]

**Pull request merged**